EPRN.PRNS.API/Repositories/Repository.cs

```diff
@@ -151,5 +151,67 @@ public async Task<StatusAndProducerDto> GetStatusAndRecipient(int id)
             await _prnContext.AddAsync(historyRecord);
             await _prnContext.SaveChangesAsync();
         }
+
+        public async Task<SentPrnsDto> GetSentPrns(GetSentPrnsDto request)
+        {
+            var recordsPerPage = request.PageSize;
+            var prns = _prnContext.PRN
+                .Include(repo => repo.WasteType)
+                .Include(repo => repo.PrnHistory)
+                .AsQueryable();
+           
+            if (!string.IsNullOrWhiteSpace(request.FilterBy))
+            {
+                var filterByStatus = (Common.Enums.PrnStatus)Enum.Parse(typeof(Common.Enums.PrnStatus), request.FilterBy);
```

Comment:
This could have been sent acros as an enum instead of a string. NET is very good at binding and will work the enums out correctly

Reply:
It comes off the select list as a string, and is stored in the db as a string,, approve for now and I will add it to a couple of optimisations I have in progress